### PR TITLE
Prebid Core: add X-Prebid header to ajax requests 

### LINKED ISSUE #6627: Prebid version in an HTTP header
## Type of issue
Proposed Enhancement

## Description
There's interest from some bidders to be able to cleanly distinguish the source of traffic. Currently it's up to each bidder to read the Prebid.js version and pass it to their endpoint. There aren't any conventions about where bidders should place this version, so they've done it every which way. There are 83 bid adapters that refer to prebid.version, with not much overlap in what they call the attribute: prebidVersion, pbjs, wrapperVersion, wrapper_version, version, pbav, hb_version, pubpbv, prebidJsVersion, pub_pbver, lib_v, tk_flint, etc.

The proposal is to update ajax.js to add a new HTTP header `X-Prebid` which carries the PBJS and PBS versions:

```
X-Prebid: pbjs/4.35
```

Prebid Server would add it's own version on to this header:
```
X-Prebid: pbjs/4.35,pbs-java/1.61
```

It's recognized that it may take a while before the majority of publishers have upgraded to the version of PBJS that contains this header, but you have to start somewhere.

### PR DESCRIPTION
**Pull request by snapwich**
## Type of change
- [x] Feature

## Description of change
Adds `X-Prebid` header to ajax requests.

## Other information
From issue #6627

### CODE REVIEW

**Review by patmmccann — APPROVED** [review]

File: src/ajax.js [hunk 1]
```diff
@@ -75,6 +75,7 @@ export function ajaxBuilder(timeout = 3000, {request, done} = {}) {
       if (options.withCredentials) {
         x.withCredentials = true;
       }
+      x.setRequestHeader('X-Prebid', 'pbjs/$prebid.version$');
```
Comment (bretg) [inline src/ajax.js:78]: Sorry for the change here @snapwich , but the PBS committee today noted that the prefered HTTP header syntax is lower case. May I ask you to change this to "x-prebid" please?

**Review by bretg — CHANGES_REQUESTED** [review]

**Comment by harpere** [comment]
Also @snapwich - There's a strange Circle CI error.  Any idea what this is from?  That looks like the hash of your commit.
`fatal: reference is not a tree: 827c941e08218c55f7053782e6272f76f11c26c7`

File: src/ajax.js (hunk 1 above)
Comment (snapwich) [reply src/ajax.js:78]: i'll change the `X-Requested-With` below it to be lowercase as well just to be consistent.

**Comment by snapwich** [comment]
@harpere no clue where that error is coming from. hopefully pushing a new commit fixes it.

**Review by bretg — APPROVED** [review]

**Pull request merged**